src/posit/connect/tasks.py

```diff
@@ -0,0 +1,204 @@
+
+        The process output produced by the task.
+
+        Returns
+        -------
+        List[str]
+        """
+        return self["output"]
+
+    @property
+    def error_code(self) -> int | None:
+        """The error code.
+
+        The error code produced by the task. A non-zero value represent an
+        error. A zero value represents no error.
+
+        Returns
+        -------
+        int | None
+            Non-zero value indicates an error.
+        """
+        return self["code"] if self.is_finished else None
+
+    @property
+    def error_message(self) -> str | None:
+        """The error message.
+
+        Returns
+        -------
+        str | None
+            Human readable error message, or None on success or not finished.
+        """
+        return self.get("error") if self.is_finished else None
+
+    @property
+    def result(self) -> dict | None:
+        """The task result.
+
+        Returns
+        -------
+        dict | None
+        """
+        return self.get("result")
+
+    # CRUD Methods
+
+    @overload
+    def update(self, first: int, wait: int, **kwargs) -> None:
+        """Update the task.
+
+        Parameters
+        ----------
+        first : int, default 0
+            Line to start output on.
+        wait : int, default 0
+            Maximum number of seconds to wait for the task to complete.
+        """
+        ...
+
+    @overload
+    def update(self, *args, **kwargs) -> None:
+        """Update the task."""
+        ...
+
+    def update(self, *args, **kwargs) -> None:
+        """Update the task.
+
+        See Also
+        --------
+        task.wait_for : Wait for the task to complete.
+
+        Notes
+        -----
+        When waiting for a task to complete, one should consider utilizing `task.wait_for`.
+
+        Examples
+        --------
+        >>> task.output
+        [
+            "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
+        ]
+        >>> task.update()
+        >>> task.output
+        [
+            "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
+            "Pretium aenean pharetra magna ac placerat vestibulum lectus mauris."
+        ]
+        """
+        params = dict(*args, **kwargs)
+        path = f"v1/tasks/{self.id}"
+        url = urls.append(self.config.url, path)
+        response = self.session.get(url, params=params)
+        result = response.json()
+        super().update(**result)
+
+    def wait_for(self, sleep: int = 1) -> None:
+        """Wait for the task to finish.
+
+        Parameters
+        ----------
+        sleep : int, optional
+            Maximum number of seconds to wait between status checks.
+
+        Examples
+        --------
+        >>> task.wait_for()
+        None
+
+        Wait five seconds between status checks.
+        >>> task.wait_for(5)
```

Comment:
When I read this, I thought this would wait for 5 seconds and then return. But it turns out that it will wait as long as it takes to finish, just updating at 5-second intervals.

Reply:
It does read that way. I'm going to remove the argument for now an address it in #184